File: 14 - Find-Three-Largest-Numbers/14.js
function findThreeLargestNumbers(array) {
    let lowest = Number.NEGATIVE_INFINITY;
    let medium = Number.NEGATIVE_INFINITY;
    let largest = Number.NEGATIVE_INFINITY;

    for (const elem of array) {
        if (elem >= lowest) {
            lowest = elem;

            if (lowest >= medium) {
                [lowest, medium] = [medium, lowest];
            }

            if (medium >= largest) {
                [largest, medium] = [medium, largest];
            }

            if (lowest >= medium) {
                [lowest, medium] = [medium, lowest];
            }
        }
    }

    return [lowest, medium, largest];
}

File: easy/14 - Find-Three-Largest-Numbers/14.py
def find_three_largest_numbers(array):
    lowest = medium = largest = float("-inf")

    for elem in array:
        if elem >= lowest:
            lowest = elem

            if lowest >= medium:
                lowest, medium = medium, lowest

            if medium >= largest:
                largest, medium = medium, largest

            if lowest >= medium:
                lowest, medium = medium, lowest

    return [lowest, medium, largest]
